loadbalancing.md

```diff
@@ -0,0 +1,345 @@
+
+Some of the newer MySQL connectors have support for [DNS SRV](https://dev.mysql.com/doc/refman/8.0/en/connecting-using-dns-srv.html). This is somewhat similar to round-robin DNS, but allows you to set priorities. The benefit from this is that it is an industry standard and that this makes the client-side configuration easier.
+
+If you are using Kubernetes you probably want to use the [LoadBalancer](https://docs.pingcap.com/tidb-in-kubernetes/stable/access-tidb#loadbalancer) service type that works with Amazon ELB and similar services from other cloud vendors. For on-premise Kubernetes something like [metallb](https://metallb.universe.tf/) can be used.
+
+When your are not using Kubernetes to deploy on a cloud service you can still use the loadbalancing service from your cloud provider.
+
+Another common option is to use a software based loadbalancer. Common ones are ProxySQL, haproxy and MySQL Router.
+
+The last type of loadbalancer that we will discuss here is hardware based loadbalancers. These are physical machines where you plug in a network cable. These are often costly, but also offer a high throughput.
+
+## General loadbalancer requirements for TiDB
+
+Where loadbalancers may offer advanced features like read/write splitting for MySQL, this is not needed for TiDB. The loadbalancer does not need to be able to inspect the MySQL protocol. A TCP level loadbalacer works fine for TiDB.
+
+The loadbalancer only needs access to the TiDB servers (TCP port 4000 by default) and doesn't need access to PD or TiKV servers. When using the HTTP status access to TCP port 10080 is also required.
+
+It is recommended to use `http://<ip_of_tidb>:10080/status` as the health check since it supports the use of the _graceful shutdown_ feature. This feature allows you to first drain client connections before shutting down a TiDB Server and reduces the impact on applications. The [`graceful-wait-before-shutdown`](/tidb-configuration-file.md#graceful-wait-before-shutdown-new-in-v50) variable needs to be set to a non-zero value to benefit from this feature.
+
+Health checks that check for `read_only` or other MySQL or InnoDB variables should be avoided.
+
+## Where to put your loadbalancer
+
+For software based loadbalancers you have the option to put your loadbalancer on the same hosts as your application or on separate hosts. Both have their own benefits.
+
+Putting the loadbalancer on the same hosts as your application is good for performance as it doesn't need an extra network hop. However having to adminster many instances of your loadbalancing software makes this slightly more complicated to administer.
+
+Putting the loadbalancer on separate machines may be less performant as it needs an extra network hop and depending on your traffic may become a choke-point. However for administration this is might be easier.
```

Comment:
Performant is kind of jargon. It is well understood but I don't think we should use it in docs.

The style-guide is also to s/may/might/ because may has multiple meanings. But I think in this context we could even say something like "will typically increase network latency" because I think it is unequivocal? The edge case would be something weird like a routing error (I also changed it to latency, since there is no guarantee throughput will drop, but it probably will accordingly for many workloads because the protocol is synchronous).

s/machines/hosts/ as physical is not implied, and host is used earlier.

```suggestion
Installing the loadbalancer on separate hosts will typically increase latency as an extra network hop is required. There is also risk that (depending on your network) connectivity to and from the load balancer could become saturated. However, this method is less complicated to administer.
```